core/src/test/scala/unit/kafka/log/LogCleanerTest.scala

```diff
@@ -62,6 +65,33 @@ class LogCleanerTest {
     Utils.delete(tmpdir)
   }
 
+  @Test
+  def testRemoveMetricsOnClose(): Unit = {
+    val mockMetricsGroupCtor = mockConstruction(classOf[KafkaMetricsGroup])
+    try {
+      val logCleaner = new LogCleaner(new CleanerConfig(true),
+        logDirs = Array(TestUtils.tempDir()),
+        logs = new Pool[TopicPartition, UnifiedLog](),
+        logDirFailureChannel = new LogDirFailureChannel(1),
+        time = time)
+
+      // shutdown logCleaner so that metrics are removed
+      logCleaner.shutdown()
+
+      val mockMetricsGroup = mockMetricsGroupCtor.constructed.get(0)
+      val numMetricsRegistered = 5
+      verify(mockMetricsGroup, times(numMetricsRegistered)).newGauge(anyString(), any())
+      verify(mockMetricsGroup, times(numMetricsRegistered)).removeMetric(anyString())
```

Comment:
nit: Should we actually verify that `removeMetric` was called for the metric name that we expected?

Reply:
The latest revision contains the changes that you have suggested.